bitshuffle/bitshuffle.py

```diff
@@ -300,6 +319,62 @@ def decode(message):
         return payload, checksum_ok
 
 
+def infer_mode(args):
+    """If encode/decode is unknown, return which it 'should' be,
+    based on various heuristics.
+    TODO: return error if both encode and decode are specified
+    """
+    if args.encode or args.decode:
+        return args
+
+    elif any((args.compresstype, args.compresslevel,
+              args.chunksize, args.filename)):
+        args.encode = True
+
+    # this is a submenu: could specify editor to compose
+    # TODO: check editor when encoding as well as decoding
+
+    elif args.editor:
+        args.decode = True
+
+    elif args.input:
+        if args.output:
+            args.decode = True
+        else:
+            args.encode = True
+
+    elif stdin.isatty():
+        args.decode = True
+
+    elif not args.output:
+        args.encode = True
+
+    elif args.output and not stdin.isatty():
+        args.decode = True
+
+    else:
+        parser.print_help()
+        quit(1)
+
+    return args
+
+
+def set_defaults(args):
+
```

Comment:
My only concern for this is updating the defaults is no longer automatic. If we change input and output as per https://github.com/charlesdaniels/bitshuffle/issues/28, we'll have to update the help manually. Same for any other changes. Not sure how to fix this.